src/core/filters/packets/filter.rs

```diff
@@ -23,10 +39,79 @@ impl FilterPacket {
         let insns = program.get_instructions();
         let filter = BpfProg::try_from(unsafe { mem::transmute::<_, &[u8]>(insns) })?;
 
-        Ok(FilterPacket(eBpfProg::try_from(filter)?))
+        let ebpf_filter = eBpfProg::try_from(filter)?;
+        if ebpf_filter.len() > FILTER_MAX_INSNS {
+            bail!("Filter exceeds the maximum allowed size.");
+        }
+
+        Ok(FilterPacket(ebpf_filter))
     }
 
     pub(crate) fn to_bytes(&self) -> Result<Vec<u8>> {
         Ok(self.0.to_bytes())
     }
 }
+
+pub(crate) unsafe extern "C" fn prepare_load(
+    prog: *mut libbpf_sys::bpf_program,
+    _opts: *mut libbpf_sys::bpf_prog_load_opts,
+    cookie: ::std::os::raw::c_long,
+) -> std::os::raw::c_int {
+    let filter = get_filter(cookie as u32);
+
+    let f = if let Some(f) = filter {
+        match f {
+            Filter::Packet(bf) => bf.0,
+        }
+    } else {
+        let mut default_filter = eBpfProg::new();
+        default_filter.add(eBpfInsn::mov32(MovInfo::Imm {
+            dst: BpfReg::R0,
+            imm: 0x40000_i32,
+        }));
+        default_filter.to_bytes()
+    };
+
+    let filter: &[libbpf_sys::bpf_insn] = unsafe {
+        std::slice::from_raw_parts(
+            f.as_slice().as_ptr() as *const libbpf_sys::bpf_insn,
+            f.len() / std::mem::size_of::<libbpf_sys::bpf_insn>(),
+        )
+    };
+
+    let (insns, insns_cnt) = unsafe {
+        (
+            libbpf_sys::bpf_program__insns(prog),
+            libbpf_sys::bpf_program__insn_cnt(prog),
+        )
+    };
+
+    let insns = unsafe { std::slice::from_raw_parts(insns, insns_cnt as usize) };
+    let mut prog_ext = insns.to_vec().clone();
+    let mut inject = None;
+
+    for (pos, insn) in prog_ext.iter().enumerate() {
```

Comment:
nit: I think you could replace the variable, loop and unwrap for a my rust-y way:

```rust
let inject_pos = match prog_ext.iter().enumerate().filter_map(|(pos, ins)| (ins.imm == cookie).then(|| pos)).next() {
    Some(p) => p,
    None => {
          warn!();
          exit 1;
    }
}
```